packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx

```diff
@@ -173,6 +174,10 @@ const TinyMCE = (props: NoteBodyEditorProps, ref: any) => {
 		}, 10);
 	};
 
+	useEffect(() => {
+		setLeftExtraToolbarWidth(document.getElementById('leftExtraToolbarContainer').offsetWidth);
+	}, [props.noteToolbarButtonInfos]);
+
```

Comment:
> Because your solution has a race condition if leftExtraToolbarContainer is updated with a delay, then you'll read the previous width, not the current one.


@laurent22  Here I have a different thought, that code inside `useEffect` will run after the document is rendered again. So if there is any  change in the title, the `TinyMCE` component will rerender and after it is rendered completely, then the `useEffect` will run. 

And if I am not wrong, if somehow leftExtraToolbarContainer is updated with a delay, the `useEffect` will run after it is rendered not before that. So, it would be great if you kindly explain such a case in little more detail if I am incorrect anywhere.

> Is it not possible to listen to an event when the left toolbar size changes?

Surely, there must be one way. But I think( though not sure as I still don't know how to do it) it would get complicated.

Reply:
Thanks again, much more clear now...